### PR TITLE
Fix Reactive PG and MySQL Clients initialization

### PR DESCRIPTION
**Pull request by mkouba**
- make sure the PgPoolProducer is initialized before the StartupEvent is
fired

See the zulip conversation: https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/reactive-pg-client

### CODE REVIEW

File: extensions/reactive-pg-client/deployment/src/main/java/io/quarkus/reactive/pg/client/deployment/ReactivePgClientProcessor.java [hunk 1]
```diff
@@ -37,7 +38,8 @@ NativeImageConfigBuildItem config() {
 
     @BuildStep
     @Record(ExecutionTime.RUNTIME_INIT)
-    void build(BuildProducer<FeatureBuildItem> feature, BuildProducer<PgPoolBuildItem> pgPool, PgPoolRecorder recorder,
+    ServiceStartBuildItem build(BuildProducer<FeatureBuildItem> feature, BuildProducer<PgPoolBuildItem> pgPool,
+            PgPoolRecorder recorder,
```
Comment (gsmet) [inline extensions/reactive-pg-client/deployment/src/main/java/io/quarkus/reactive/pg/client/deployment/ReactivePgClientProcessor.java:42]: Shouldn't we do the same in `ReactiveMySQLClientProcessor` ?

**Review by gsmet — CHANGES_REQUESTED** [review]

File: extensions/reactive-pg-client/deployment/src/main/java/io/quarkus/reactive/pg/client/deployment/ReactivePgClientProcessor.java (hunk 1 above)
Comment (mkouba) [reply extensions/reactive-pg-client/deployment/src/main/java/io/quarkus/reactive/pg/client/deployment/ReactivePgClientProcessor.java:42]: IDK. I was asked to fix this one ;-). But if it's using the same pattern then yes... I'll take a look.

**Review by cescoffier — APPROVED** [review]

**Comment by mkouba** [comment]
Hm, `JDK 11 JVM Tests` fails with:
```
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 5.936 s <<< FAILURE! - in io.quarkus.smallrye.reactivemessaging.amqp.devmode.AmqpDevModeTest
[ERROR] testCodeUpdate  Time elapsed: 5.514 s  <<< ERROR!
java.lang.IllegalStateException: Arc context propagation was attempted but the container is not running.
	at io.quarkus.arc.runtime.context.ArcContextProvider.lambda$currentContext$2(ArcContextProvider.java:35)
	at io.smallrye.context.impl.ActiveContextState.<init>(ActiveContextState.java:18)
	at io.smallrye.context.impl.CapturedContextState.begin(CapturedContextState.java:35)
	at io.smallrye.context.SmallRyeThreadContext.lambda$withContext$0(SmallRyeThreadContext.java:213)
	at io.smallrye.context.propagators.rxjava2.ContextPropagatorOnFlowableCreateAction$ContextCapturerFlowable.onNext(ContextPropagatorOnFlowableCreateAction.java:50)
	at io.reactivex.internal.operators.flowable.FlowableFlatMap$MergeSubscriber.drainLoop(FlowableFlatMap.java:506)
	at io.reactivex.internal.operators.flowable.FlowableFlatMap$MergeSubscriber.drain(FlowableFlatMap.java:371)
```
Does anyone have a clue what happened?

**Comment by geoand** [comment]
> Hm, `JDK 11 JVM Tests` fails with:
> 
> ```
> [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 5.936 s <<< FAILURE! - in io.quarkus.smallrye.reactivemessaging.amqp.devmode.AmqpDevModeTest
> [ERROR] testCodeUpdate  Time elapsed: 5.514 s  <<< ERROR!
> java.lang.IllegalStateException: Arc context propagation was attempted but the container is not running.
> 	at io.quarkus.arc.runtime.context.ArcContextProvider.lambda$currentContext$2(ArcContextProvider.java:35)
> 	at io.smallrye.context.impl.ActiveContextState.<init>(ActiveContextState.java:18)
> 	at io.smallrye.context.impl.CapturedContextState.begin(CapturedContextState.java:35)
> 	at io.smallrye.context.SmallRyeThreadContext.lambda$withContext$0(SmallRyeThreadContext.java:213)
> 	at io.smallrye.context.propagators.rxjava2.ContextPropagatorOnFlowableCreateAction$ContextCapturerFlowable.onNext(ContextPropagatorOnFlowableCreateAction.java:50)
> 	at io.reactivex.internal.operators.flowable.FlowableFlatMap$MergeSubscriber.drainLoop(FlowableFlatMap.java:506)
> 	at io.reactivex.internal.operators.flowable.FlowableFlatMap$MergeSubscriber.drain(FlowableFlatMap.java:371)
> ```
> 
> Does anyone have a clue what happened?

I have seen that in a lot of PRs.... I have no idea what the deal with it is.

**Comment by gsmet** [comment]
@cescoffier is aware of the issue.

**Review by gsmet — APPROVED** [review]

**Pull request merged**